### PR TITLE
monitoring: grafana ux improvements

### PR DESCRIPTION
**Pull request by bobheadxi**
- Refreshed homepage
   - **everything now links to service dashboards** - the warning/critical graphs, the table, etc. Just click!
   - upgraded table to get data links on the table cells. each table cell goes direct to the relevant graph panel
   - upgraded homepage graph to timeseries to avoid the annoying annotation prompt (https://github.com/grafana/grafana/issues/25580#issuecomment-805731081) when clicking to get to the link - doing this for the other dashboards was too hard though due to lack of support in the library, so those still use the old 
 - Refreshed container overviews
   - **alerts panels now links to relevant graph panels** - just click!
   - upgraded tables similar to above, table cells link to graph panels

Closes https://github.com/sourcegraph/sourcegraph/issues/16940

Also note that each panel already links to alert solutions and dashboard docs :)

image

image

Try it out:

```
kubectl port-forward svc/prometheus 9090:30090
sg run grafana
```

### CODE REVIEW

**Comment by slimsag** [comment]
very nice

File: monitoring/monitoring/internal/grafana/alerts-defined.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package grafana
+
+import "github.com/grafana-tools/sdk"
+
+func NewContainerAlertsDefinedTable(target sdk.Target) *sdk.Panel {
+	alertsDefined := sdk.NewCustom("Alerts defined")
+	alertsDefined.Type = "table"
+
+	var panelTemplateLink = "/-/debug/grafana/d/${__data.fields.service_name}/${__data.fields.service_name}?viewPanel=${__data.fields.grafana_panel_id}"
+	alertsDefined.CustomPanel = &sdk.CustomPanel{
```
Comment (bobheadxi) [inline monitoring/monitoring/internal/grafana/alerts-defined.go:10]: This is a bit dirty but IMO not really worth investing more in, this is only to get the data links

File: monitoring/monitoring/internal/grafana/alerts-defined.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+							Title: "Graph panel",
+							URL:   &panelTemplateLink,
+						}}),
+					},
+				},
+				alertsFiringOverride(),
+				{
+					Matcher: matcherByName("grafana_panel_id"),
+					Properties: []OverrideProperty{
+						propertyWidth(0.1),
```
Comment (bobheadxi) [inline monitoring/monitoring/internal/grafana/alerts-defined.go:33]: hack to hide columns while still using value for template variables (if you hide the column, it's no longer available in the template vars :( https://github.com/grafana/grafana/issues/24092)

File: go.mod [hunk 3]
```diff
@@ -339,6 +339,8 @@ replace (
 replace (
 	// Pending: https://github.com/ghodss/yaml/pull/65
 	github.com/ghodss/yaml => github.com/sourcegraph/yaml v1.0.1-0.20200714132230-56936252f152
+	// graph-fieldconfig branch - https://github.com/grafana-tools/sdk/pull/170
+	github.com/grafana-tools/sdk => github.com/sourcegraph/grafana-sdk v0.0.0-20210914025534-b4255965ba53
```
Comment (bobheadxi) [inline go.mod:343]: https://github.com/grafana-tools/sdk/pull/170

**Review by coury-clark — APPROVED** [review]
Nice :)

File: go.mod (hunk 3 above)
Comment (slimsag) [reply go.mod:342]: Make sure we keep an eye on the license of this SDK when updating BTW (I just confirmed, it's still apache)

**Comment by slimsag** [comment]
seriously, very cool

**Comment by justdueck** [comment]
this is awesome. thanks @bobheadxi!

File: go.mod (hunk 3 above)
Comment (bobheadxi) [reply go.mod:342]: Yep! It's not an official project though (it's really out of date as indicated by this PR 🥲 ) so I think we're safe from a license change

**Pull request merged**